src/io/parquet/geoparquet_metadata.rs

```diff
@@ -43,3 +46,73 @@ impl GeoParquetMetadata {
         ))
     }
 }
+
+// TODO: deduplicate with `resolve_types` in `downcast.rs`
+fn infer_geo_data_type(
+    geometry_types: &HashSet<&str>,
+    coord_type: CoordType,
+) -> Result<Option<GeoDataType>> {
+    if geometry_types.iter().any(|t| t.contains(" Z")) {
+        return Err(GeoArrowError::General(
+            "3D coordinates not currently supported".to_string(),
+        ));
+    }
+
+    match geometry_types.len() {
+        0 => Ok(None),
+        1 => Ok(Some(match *geometry_types.iter().next().unwrap() {
+            "Point" => GeoDataType::Point(coord_type),
+            "LineString" => GeoDataType::LineString(coord_type),
+            "Polygon" => GeoDataType::Polygon(coord_type),
+            "MultiPoint" => GeoDataType::MultiPoint(coord_type),
+            "MultiLineString" => GeoDataType::MultiLineString(coord_type),
+            "MultiPolygon" => GeoDataType::MultiPolygon(coord_type),
+            "GeometryCollection" => GeoDataType::GeometryCollection(coord_type),
+            _ => unreachable!(),
+        })),
+        2 => {
+            if geometry_types.contains("Point") && geometry_types.contains("MultiPoint") {
+                Ok(Some(GeoDataType::MultiPoint(coord_type)))
+            } else if geometry_types.contains("LineString")
+                && geometry_types.contains("MultiLineString")
+            {
+                Ok(Some(GeoDataType::MultiLineString(coord_type)))
+            } else if geometry_types.contains("Polygon") && geometry_types.contains("MultiPolygon")
+            {
+                Ok(Some(GeoDataType::MultiPolygon(coord_type)))
+            } else {
+                Ok(Some(GeoDataType::Mixed(coord_type)))
+            }
+        }
+        _ => Ok(Some(GeoDataType::Mixed(coord_type))),
+    }
+}
+
+pub fn parse_geoparquet_metadata(
```

Comment:
Note that this has turned from a private to a public function.

Reply:
Nevermind, I've turned it back to a private function in deb4c32830eef030d5159b501c6bbac44cc7fc69.